base/show.jl

```diff
@@ -324,6 +324,15 @@ short
 """
 IOContext(io::IO, KV::Pair, KVs::Pair...) = IOContext(IOContext(io, KV), KVs...)
 
+"""
+    push(io::IO, KV::Pair...)
+
+Create an `IOContext` that wraps a given stream, adding the specified `key=>value` pairs to
+the properties of that stream (note that `io` can itself be an `IOContext`).
+This is equivalent to [`IOContext(io, KV...)`](@ref).
```

Comment:
Whats the advantage of `push` then? Personally I would prefer reading code that had `IOContext(io, kv...)`. `push(io, ...)` feels like it should write something to the `io`, not returning an io with modified context.

Reply:
Alternative API might be `Base.setindex(io, V, K)` and/or `merge(io, (K1 = V1, K2 = V2))`.

Reply:
I agree that `push` is less convincing for `IOContext` than for `ImmutableDict`. Still, `IO` types already have the "dictionary" semantics and `getindex`. But `IO` is special enough compared to "normal" collections that I see this can be more controversial to add `push` for it. Seems like I will have to shrink this shrinked PR even more ;-)